molo/profiles/views.py

```diff
@@ -9,24 +8,27 @@
 from django.shortcuts import render
 from django.contrib import messages
 from django.contrib.auth.models import User
+from django.contrib.auth import authenticate, login
 from django.utils.translation import ugettext_lazy as _
 
 
-@csrf_protect
-def register(request):
-    if request.method == 'POST':
-        form = RegistrationForm(request.POST)
-        if form.is_valid():
-            user = User.objects.create_user(
-                username=form.cleaned_data['username'],
-                password=form.cleaned_data['password'],
-            )
-            user.profile.date_of_birth = form.cleaned_data['date_of_birth']
-            user.profile.save()
-            return HttpResponseRedirect(form.cleaned_data.get('next', '/'))
-        return render(request, 'profiles/register.html', {'form': form})
-    form = RegistrationForm()
-    return render(request, 'profiles/register.html', {'form': form})
```

Comment:
Yes it does. All that crap is handled by the Class based view. You can [learn more](https://docs.djangoproject.com/en/1.8/topics/class-based-views/generic-editing/) about it here.